src/ray/raylet/worker_pool.cc

```diff
@@ -392,6 +425,105 @@ Process WorkerPool::StartProcess(const std::vector<std::string> &worker_command_
   return child;
 }
 
+Process WorkerPool::StartContainerProcess(
```

Comment:
we don't need this function to be put in CPP layer. Shim process should be good enough. 

Check it here https://github.com/ray-project/ray/blob/master/python/ray/workers/setup_runtime_env.py

Right now it only take care of conda env, like install conda and activate and start default worker. 

I think you can process it there in the same way, but make sure run it before the conda env get processed.

@architkulkarni  could you help double-check it's working with shim process?